### DISCUSSION TITLE
Lets talk about Reinforcement learning

### DISCUSSION THREAD

**Discussion by JulioJerez** [JulioJerez/newton-dynamics#51] [General]
I’m opening this thread to casually discuss reinforcement learning and exchange ideas and perspectives.

As many of you probably seen, some time ago I started building my own neural network library, primarily focused on continuous-action algorithms for controlling rigid body systems.

My main focus is to develop a version of the two most popular methods for continue actions RL:
-_Soft actor critics (SAC)_
-_Proximal Policy Gradient (PPO)_

Looking back, that may have been a mistake. 
In hindsight, I probably should have just used PyTorch or TensorFlow. 
After experimenting with both, I quickly ruled out TensorFlow for several reasons. 
I then spent some time working with PyTorch, but eventually I began to question whether calling it “open source” is entirely accurate. 
It feels more like open source with significant strings attached.

That said, one of my main concerns with the current state of reinforcement learning isn’t just tooling. 
It’s how loosely many tech companies use mathematical terminology, sometimes to the point where it feels misleading. 
There seems to be a pervasive culture of renaming and reinventing existing ideas, which can create the impression that entirely new branches of mathematics or statistics are being invented. 
In other cases, using some idea like a probability density as a substitute for a probability. 
This seem particularly strong with the Pytorch and OpenAI community. 
More on that later.
